string detector_ns
time stamp